requirements/py311.yml

```diff
@@ -14,10 +14,10 @@ dependencies:
   - cartopy >=0.21
   - cf-units >=3.1
   - cftime >=1.5
-  - dask-core >=2025.1.0  # github.com/SciTools/iris/issues/6264
+  - dask-core >=2025.1.0,<2025.10.0  # github.com/SciTools/iris/issues/6752
```

Comment:
We cannot use the lockfile PR as the 'record' of this. The automation freely overwrites the contents every week, and we are explicitly expecting this to happen as soon as #6773 is merged.

We need issues that explicitly state "remove X pin", together with any other details. This is especially true of Dask since it looks like @rcomer has done some investigation.